internal/nonkube/common/systemd.go

```diff
@@ -260,3 +272,50 @@ func IsLingeringEnabled(user string) bool {
 	_, err := os.Stat(lingerFile)
 	return err == nil
 }
+
+func NewSystemdGlobal(platform string) (SystemdGlobal, error) {
+
+	return &systemdGobal{
+		getUid:              os.Getuid,
+		command:             exec.Command,
+		rootSystemdBasePath: rootSystemdBasePath,
+		platform:            platform,
+	}, nil
+}
+
+func (sg *systemdGobal) getCmdEnablePodmanSocket() *exec.Cmd {
+	return sg.command("systemctl", "--user", "enable", "--now", "podman.socket")
```

Comment:
The systemd commands must take into account the current user id.
If `getUid()` is 0, then do not include the `--user` argument (for all commands).